php-frontend/src/main/java/org/sonar/php/tree/impl/statement/ForStatementTreeImpl.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl.statement;
+
+import com.google.common.base.Functions;
+import com.google.common.collect.Iterators;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.php.tree.impl.SeparatedList;
+import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.ForStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.visitors.TreeVisitor;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+public class ForStatementTreeImpl extends PHPTree implements ForStatementTree {
+
+  private final Kind KIND;
+
+  private final ForStatementHeader header;
+  private final InternalSyntaxToken colonToken;
+  private final List<StatementTree> statements;
+  private final InternalSyntaxToken endForToken;
+  private final InternalSyntaxToken eosToken;
+
+  private ForStatementTreeImpl(
+      Kind kind,
+      ForStatementHeader header,
+      @Nullable InternalSyntaxToken colonToken, List<StatementTree> statements,
+      @Nullable InternalSyntaxToken endForToken, @Nullable InternalSyntaxToken eosToken
+  ) {
+    this.header = header;
+    this.colonToken = colonToken;
+    this.endForToken = endForToken;
+    this.statements = statements;
+    this.eosToken = eosToken;
+
+    this.KIND = kind;
+  }
+
+  public ForStatementTreeImpl(ForStatementHeader header, InternalSyntaxToken colonToken, List<StatementTree> statements, InternalSyntaxToken endForToken, InternalSyntaxToken eosToken) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 185 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)